src/widgets/modals/GenericChooseModal.js

```diff
@@ -0,0 +1,86 @@
+/* eslint-disable react/forbid-prop-types */
+/**
+ * mSupply Mobile
+ * Sustainable Solutions (NZ) Ltd. 2019
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { FlatList, StyleSheet, Text, TouchableOpacity, View } from 'react-native';
+
+import { PageContentModal } from './PageContentModal';
+
+import { COMPONENT_HEIGHT } from '../../globalStyles';
+
```

Comment:
Done 91c8b7c9feb4dc51dd412505769a19a7ce3790a5

I was considering putting details of how it could be improved in the future. EG using `extraData`, accepting an array of strings rather than objects but not sure how I feel about those sorts of comments.